src/Appwrite/Realtime/Realtime.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace Appwrite\Realtime;
+
+use Appwrite\Auth\Auth;
+use Appwrite\Database\Document;
+
+class Realtime
+{
+    /**
+     * @var Document $user
+     */
+    static $user;
+
+    /**
+     * @param Document $user
+     */
+    static function setUser(Document $user)
+    {
+        self::$user = $user;
+    }
+
+    /**
+     * @return array
+     */
+    static function getRoles()
+    {
+        $roles = ['*', 'role:' . ((self::$user->isEmpty()) ? Auth::USER_ROLE_GUEST : Auth::USER_ROLE_MEMBER)];
```

Comment:
I would love to think of a way to leverage and reuse Authorization::getRoles for this. Might be a bit complex because some of the roles are being dynamically added on the REST API init and some on the user resource creation, but maybe we can do some refactoring. We can discuss more later.

Reply:
Gonna look into it over the week 👍🏻